### PR TITLE
 transaction-isolation-levels.md: update txn isolation

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What is changed, added or deleted? (Required)

Update TiDB transaction-isolation document, make it compatible with current TiDB version(master branch and release-4.0)

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: transaction-isolation-levels.md [hunk 1]
```diff
@@ -22,7 +22,9 @@ TiDB 实现了快照隔离 (Snapshot Isolation, SI) 级别的一致性。为与
 
 > **注意：**
 >
-> 在 TiDB v3.0 中，事务的自动重试功能默认为禁用状态。关于该项功能对隔离级别的影响以及如何开启该项功能，请参考[事务重试](/optimistic-transaction.md#重试机制)。
+> 在 TiDB v3.0 中，事务的自动重试功能默认为禁用状态。开启自动重试可能导致**事务隔离级别发生变化**，不建议启用，更多关于事务自动重试的文档说明，请参考[事务重试](/optimistic-transaction.md#重试机制)。
```
Comment (jackysp) [inline transaction-isolation-levels.md:25]:
```suggestion
> 在 TiDB v3.0 中，事务的自动重试功能默认为禁用状态。开启自动重试可能导致**事务隔离级别遭到破坏**，不建议启用，更多关于事务自动重试的文档说明，请参考[事务重试](/optimistic-transaction.md#重试机制)。
```

**Comment by yikeke** [comment]
Because this week is Document Special Week, the merge button is temporarily disabled, so we can only merge this PR until the end of the Special Week. Is it acceptable for you? @cfzjywxk @jackysp

**Comment by cfzjywxk** [comment]
> Because this week is Document Special Week, the merge button is temporarily disabled in the docs-cn repo, so we can only merge this PR until the end of the Special Week (presumably this Friday night). Is it acceptable for you? @cfzjywxk @jackysp

It's ok

File: transaction-isolation-levels.md [hunk 2]
```diff
@@ -38,11 +40,12 @@ insert into t1 values(0);
...
 start transaction;              |               start transaction;
 select * from t1;               |               select * from t1;
-update t1 set id=id+1;          |               update t1 set id=id+1;
+update t1 set id=id+1;          |               update t1 set id=id+1; -- 如果使用悲观事务，则后一个执行的 update 语句会等锁，直到持有锁事务提交或者回滚释放行锁
 commit;                         |
-                                |               commit; -- 事务提交失败，回滚
+                                |               commit; -- 事务提交失败，回滚。如果使用悲观事务，可以提交成功
 ```
 
+
```
Comment (yikeke) [inline transaction-isolation-levels.md:48]:
```suggestion
```
To fix CI

File: transaction-isolation-levels.md [hunk 3]
```diff
@@ -22,7 +22,9 @@ TiDB 实现了快照隔离 (Snapshot Isolation, SI) 级别的一致性。为与
 
 > **注意：**
 >
-> 在 TiDB v3.0 中，事务的自动重试功能默认为禁用状态。关于该项功能对隔离级别的影响以及如何开启该项功能，请参考[事务重试](/optimistic-transaction.md#重试机制)。
+> 在 TiDB v3.0 中，事务的自动重试功能默认为禁用状态。开启自动重试可能导致**事务隔离级别遭到破坏**，不建议启用，更多关于事务自动重试的文档说明，请参考[事务重试](/optimistic-transaction.md#重试机制)。
```
Comment (yikeke) [inline transaction-isolation-levels.md:25]:
```suggestion
> 在 TiDB v3.0 中，事务的自动重试功能默认为禁用状态。不建议开启自动重试功能，因为可能导致**事务隔离级别遭到破坏**。更多关于事务自动重试的文档说明，请参考[事务重试](/optimistic-transaction.md#重试机制)。
```

File: transaction-isolation-levels.md [hunk 4]
```diff
@@ -22,7 +22,9 @@ TiDB 实现了快照隔离 (Snapshot Isolation, SI) 级别的一致性。为与
...
+>
+> 从 TiDB [v3.0.8](/releases/release-3.0.8.md#TiDB) 版本开始，对于新创建 TiDB 集群默认使用[悲观事务模式](/pessimistic-transaction.md)，悲观事务中的当前读(for update 读)为**不可重复读**，关于悲观事务使用注意事项，请参考[悲观事务模式](/pessimistic-transaction.md)
```
Comment (yikeke) [inline transaction-isolation-levels.md:27]:
```suggestion
> 从 TiDB [v3.0.8](/releases/release-3.0.8.md#tidb) 版本开始，新创建的 TiDB 集群会默认使用[悲观事务模式](/pessimistic-transaction.md)，悲观事务中的当前读（for update 读）为**不可重复读**，关于悲观事务使用注意事项，请参考[悲观事务模式](/pessimistic-transaction.md)
```
See the [anchor link style ref](https://docs.google.com/document/d/1b6ZhZD33OoM8AacpKksGGSuxJWReLkNnSt8eSc1kTXc/edit?ts=5c9c682e#heading=h.xy6k1sq6eh08).

File: transaction-isolation-levels.md [hunk 5]
```diff
@@ -38,11 +40,12 @@ insert into t1 values(0);
 
 start transaction;              |               start transaction;
 select * from t1;               |               select * from t1;
-update t1 set id=id+1;          |               update t1 set id=id+1;
+update t1 set id=id+1;          |               update t1 set id=id+1; -- 如果使用悲观事务，则后一个执行的 update 语句会等锁，直到持有锁事务提交或者回滚释放行锁
```
Comment (yikeke) [inline transaction-isolation-levels.md:43]:
```suggestion
update t1 set id=id+1;          |               update t1 set id=id+1; -- 如果使用悲观事务，则后一个执行的 update 语句会等锁，直到持有锁的事务提交或者回滚释放行锁。
```

File: transaction-isolation-levels.md [hunk 6]
```diff
@@ -38,11 +40,12 @@ insert into t1 values(0);
...
 commit;                         |
-                                |               commit; -- 事务提交失败，回滚
+                                |               commit; -- 事务提交失败，回滚。如果使用悲观事务，可以提交成功
```
Comment (yikeke) [inline transaction-isolation-levels.md:45]:
```suggestion
                                |               commit; -- 事务提交失败，回滚。如果使用悲观事务，可以提交成功。
```

File: transaction-isolation-levels.md [hunk 7]
```diff
@@ -53,14 +56,19 @@ MySQL 可重复读隔离级别在更新时并不检验当前版本是否可见
 
 ## 读已提交隔离级别 (Read Committed)
 
-TiDB 仅在[悲观事务模式](/pessimistic-transaction.md)下支持读已提交隔离级别。在乐观事务模式下设置事务隔离级别为读已提交将不会生效，事务将仍旧使用可重复读隔离级别。
+从 TiDB [v4.0.0-beta](/releases/release-4.0.0-beta.md#TiDB) 版本开始，TiDB 支持使用 Read Committed 隔离级别。由于历史原因，当前主流数据库的读已提交隔离级别本质上都是 Oracle 定义的[一致性读隔离级别](https://docs.oracle.com/cd/B19306_01/server.102/b14220/consist.htm)。TiDB 为了适应这一历史原因，悲观事务中的 Read Committed 隔离级别的实质行为也是一致性读。
```
Comment (yikeke) [inline transaction-isolation-levels.md:59]:
```suggestion
从 TiDB [v4.0.0-beta](/releases/release-4.0.0-beta.md#tidb) 版本开始，TiDB 支持使用 Read Committed 隔离级别。由于历史原因，当前主流数据库的 Read Committed 隔离级别本质上都是 Oracle 定义的[一致性读隔离级别](https://docs.oracle.com/cd/B19306_01/server.102/b14220/consist.htm)。TiDB 为了适应这一历史原因，悲观事务中的 Read Committed 隔离级别的实质行为也是一致性读。
```
统一全文的术语用法：读已提交 -> Read Committed

File: transaction-isolation-levels.md [hunk 8]
```diff
@@ -53,14 +56,19 @@ MySQL 可重复读隔离级别在更新时并不检验当前版本是否可见
...
 
-由于历史原因，当前主流数据库的读已提交隔离级别本质上都是 Oracle 定义的一致性读隔离级别。TiDB 为了适应这一历史原因，悲观事务中的读已提交隔离级别的实质行为也是一致性读。
+> **注意：**
+>
+> Read Committed 隔离级别仅在[悲观事务模式](/pessimistic-transaction.md)下生效。在[乐观事务模式](/optimistic-transaction.md)下设置事务隔离级别为读已提交将不会生效，事务将仍旧使用可重复读隔离级别。
```
Comment (yikeke) [inline transaction-isolation-levels.md:63]:
```suggestion
> Read Committed 隔离级别仅在[悲观事务模式](/pessimistic-transaction.md)下生效。在[乐观事务模式](/optimistic-transaction.md)下设置事务隔离级别为 Read Committed 将不会生效，事务将仍旧使用可重复读隔离级别。
```

File: transaction-isolation-levels.md [hunk 9]
```diff
@@ -53,14 +56,19 @@ MySQL 可重复读隔离级别在更新时并不检验当前版本是否可见
...
 
 ### 与 MySQL 读已提交隔离级别的区别
 
 MySQL 的读已提交隔离级别大部分符合一致性读特性，但其中存在某些特例，如半一致性读 ([semi-consistent read](https://dev.mysql.com/doc/refman/8.0/en/innodb-transaction-isolation-levels.html))，TiDB 没有兼容这个特殊行为。
```
Comment (yikeke) [inline transaction-isolation-levels.md:67]:
```suggestion
MySQL 的 Read Committed 隔离级别大部分符合一致性读特性，但其中存在某些特例，如半一致性读 ([semi-consistent read](https://dev.mysql.com/doc/refman/8.0/en/innodb-transaction-isolation-levels.html))，TiDB 没有兼容这个特殊行为。
```

File: transaction-isolation-levels.md [hunk 10]
```diff
@@ -53,14 +56,19 @@ MySQL 可重复读隔离级别在更新时并不检验当前版本是否可见
...
 
-TiDB 仅在[悲观事务模式](/pessimistic-transaction.md)下支持读已提交隔离级别。在乐观事务模式下设置事务隔离级别为读已提交将不会生效，事务将仍旧使用可重复读隔离级别。
+从 TiDB [v4.0.0-beta](/releases/release-4.0.0-beta.md#TiDB) 版本开始，TiDB 支持使用 Read Committed 隔离级别。由于历史原因，当前主流数据库的读已提交隔离级别本质上都是 Oracle 定义的[一致性读隔离级别](https://docs.oracle.com/cd/B19306_01/server.102/b14220/consist.htm)。TiDB 为了适应这一历史原因，悲观事务中的 Read Committed 隔离级别的实质行为也是一致性读。
 
-由于历史原因，当前主流数据库的读已提交隔离级别本质上都是 Oracle 定义的一致性读隔离级别。TiDB 为了适应这一历史原因，悲观事务中的读已提交隔离级别的实质行为也是一致性读。
+> **注意：**
+>
+> Read Committed 隔离级别仅在[悲观事务模式](/pessimistic-transaction.md)下生效。在[乐观事务模式](/optimistic-transaction.md)下设置事务隔离级别为读已提交将不会生效，事务将仍旧使用可重复读隔离级别。
 
 ### 与 MySQL 读已提交隔离级别的区别
```
Comment (yikeke) [inline transaction-isolation-levels.md:65]:
```suggestion
### 与 MySQL Read Committed 隔离级别的区别
```

**Comment by kissmydb** [comment]
@zhenjiaogao PTAL

File: transaction-isolation-levels.md [hunk 11]
```diff
@@ -38,11 +40,12 @@ insert into t1 values(0);
```
Comment (zhenjiaogao) [inline transaction-isolation-levels.md:40]: line 35 “处于可重复读隔离级别的事务不能并发的更新同一行，当时事务提交时发现该行在该事务启动后，已经被另一个已提交的事务更新过，那么该事务会回滚并启动自动重试。示例如下：”

change to 

“处于可重复读隔离级别的事务不能并发的更新同一行，当事务提交时发现该行在该事务启动后，已经被另一个已提交的事务更新过，那么该事务会回滚并启动自动重试。示例如下：”

**Pull request merged**